lib/index.js

```diff
@@ -37,16 +37,30 @@ module.exports = class SpikeUtils {
   /**
    * Given a source file path, returns the path to the final output.
    * @param {String} file - path to source file
+   * @param {String} extension - (optinal) final file extension
    * @return {File} object containing relative and absolute paths
    */
-  getOutputPath (f) {
+  getOutputPath (f, extension) {
     let file = new File(this.conf.context, f)
 
     this.conf.spike.dumpDirs.forEach((d) => {
       const re = new RegExp(`^${d}\\${path.sep}`)
-      if (file.relative.match(re)) {
-        file = new File(this.conf.output.path, file.relative.replace(re, ''))
-      }
+      if (!file.relative.match(re))
+        return
+
+      let output = file.relative.replace(re, '')
+
+      if (!extension)
+        for (let ext in this.conf.spike.matchers)
+          if (micromatch.isMatch(output, this.conf.spike.matchers[ext])) {
+            extension = ext
+            break
+          }
+
+      if (extension)
+        output = output.replace(path.extname(output), '.'+extension)
+
+      file = new File(this.conf.output.path, output)
```

Comment:
Make sure to run `npm lint` to check for lint errors, there are few in here like multi-line if/for statements without brackets